File: OCH_Tao/ABC/B/238.py
N = int(input())
A = list(map(int,input().split()))
L = [0]
for i in range(N):
  L.append((L[-1]+A[i])%360)
else:
  L.append(360)
  L.sort(reverse=True)
print(max([L[i]-L[i+1] for i in range(N+1)]))

File: OCH_Tao/ABC/B/238.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(input[0], 10);
const A = input[1].split(" ").map(x => parseInt(x, 10));
let L = [0];
for (let i = 0; i < N; i++) {
  L.push((L[L.length - 1] + A[i]) % 360);
}
L.push(360);
L.sort((a, b) => b - a);
let cnt = [];
for (let i = 0; i < N + 1; i++) {
  cnt.push(L[i] - L[i + 1]);
}
console.log(Math.max(...cnt));